content/styleguide/styleguide.rst

```diff
@@ -0,0 +1,623 @@
+
+Likewise, coding practices and styles for XML and HTML vary widely. For the sake of clarity and consistency, samples should follow the guidelines set forth here.
+
+.. _xml-element:
+
+Element
+~~~~~~~~~~~
+
+The following piece of code represents an **element**:
+
+.. code-block:: xml
+
+  <element>
+    Some content.
+  </element>
+
+.. note:: 
+
+  An element is **not** a *block* or a *tag*.
+  
+  - *Tag* is defined below.
+  - *Block* has a specific meaning in HTML and XML templates, and should generally be avoided outside those contexts.
+
+.. _xml-tag:
+  
+Tag
+~~~~~~
+
+A **tag** is the token that begins or ends an element.
+
+.. code-block:: xml
+
+  <element>  <!-- The opening tag of this element. -->
+    Some content.
+  </element> <!-- The closing tag. -->
+  
+The word *tag* has often been used to refer to the entire element. For clarity, we will avoid that here.
+
+
+.. _xml-node:
+
+Node
+~~~~~
+
+The word *node* is often used interchangeably with *element*.
+
+For clarity, we make the following distinction:
+
+- An HTML or XML document has *elements*, not *nodes*.
+- A *node* is part of a "live" DOM tree or other dynamic representation.
+
+  - An XML or HTML element becomes an *element node* in a DOM tree.
+  - There are also other types of nodes in a DOM tree.
+
+.. _xml-attributes-values:
+
+Attributes and values
+~~~~~~~~~~~~~~~~~~~~~~~
+
+An element may have attributes. Attributes have values. Values are wrapped in straight double-quotes.
+
+.. code-block:: xml
+
+  <element attribute="value">
+    Content.
+  </element>
+  
+Other names for attributes, such as *variables* or *properties*, should be avoided.
+
+.. _xml-element-content:
+  
+Element content
+~~~~~~~~~~~~~~~~
+
+The code between the opening and closing tags of an element is the content. Content can include other elements, which are called *child elements*.
+
+.. code-block:: xml
+
+  <element>
+    Content.
+    <child-element>
+      More content.
+    </child-element>
+  </element>
+  
+When an element is empty, it can be called a *null element*.
+
+.. code-block:: xml
+
+  <null-element attribute="value" />
+
+In XML, null element tags always self-close. This is not the case in HTML. 
+
+- HTML elements that are always null (for example, `<img>`) do not need to be self-closed.
+- Empty HTML elements that normally accept content have a separate closing tag.
+
+.. code-block:: html
+
+  <img src="awesome-picture.jpeg">
+
+  <script src="some-javascript.js"></script>
+
+.. _xml-capitalization:
+  
+Capitalization
+~~~~~~~~~~~~~~~~
+
+For all HTML samples, tag names and attribute names should be ``all lowercase``. 
+
+Newly-written XML examples should also be ``all lowercase``.
+
+XML examples that show actual code generated by tools in the ODK ecosystem should replicate that code exactly, regardless of its capitalization practice.
+
+
+.. endignore
```

Comment:
We need to put some attribution in this document. The original says: "This document is licensed under a Creative Commons Attribution 4.0 International License." So we just need to put where it came from and we can reuse and modify it.